install/primehub-install

```diff
@@ -0,0 +1,1316 @@
+}
+
+disable_keycloak_ssl_required() {
+  local namespace=${PRIMEHUB_NAMESPACE}
+  info "[Config] Keycloak to support external access by HTTP"
+  kubectl -n ${namespace} exec -it keycloak-0 -- \
+  /opt/jboss/keycloak/bin/kcadm.sh config credentials \
+  --server http://localhost:8080/auth \
+  --realm master \
+  --user ${KC_USER} \
+  --password=${KC_PASSWORD}
+  kubectl -n ${namespace} exec -it keycloak-0 -- \
+  /opt/jboss/keycloak/bin/kcadm.sh update realms/master -s "sslRequired=none"
+}
+
+config::singlenode() {
+  error "[Not Support Action] config singlenode"
+}
+
+diff::singlenode() {
+  error "[Not Support Action] diff singlenode"
+}
+
+diff::primehub() {
+  info "[Check] primehub.yaml"
+  export RELEASE_NAMESPACE=${PRIMEHUB_NAMESPACE}
+  if [[ "${PRIMEHUB_CHART_PATH}" != "" ]]; then
+    export CHART_PATH=$(realpath "${PRIMEHUB_CHART_PATH}")
+  else
+    export CHART_VERSION=${PRIMEHUB_VERSION}
+  fi
+  helmfile -f ./helmfiles/primehub diff --context 2 --skip-deps --args '--disable-validation'
+}
+
+error_log_primehub() {
+  local log_name='primehub.log'
+  echo "=== Bootstrap ===" > ${log_name}
+  kubectl logs -n ${PRIMEHUB_NAMESPACE} -l component=bootstrap --tail=-1 >> ${log_name}
+  error "[Error] Generate PrimeHub error log at ${log_name}."
+}
+
+progress_primehub() {
+  sleep 10
+  until kubectl -n ${PRIMEHUB_NAMESPACE} get job/primehub-bootstrap > /dev/null 2&>1; do
+    sleep 2;
+  done
+  if kubectl -n ${PRIMEHUB_NAMESPACE} get sts/keycloak > /dev/null 2>&1 ; then
+    info "[Progress] wait for keycloak ready"
+    kubectl -n ${PRIMEHUB_NAMESPACE} rollout status sts/keycloak
+  fi
+  info "[Progress] wait for bootstrap job ready"
+  kubectl -n ${PRIMEHUB_NAMESPACE} wait --for=condition=complete job/primehub-bootstrap --timeout=1000s
+  info "[Progress] wait for graphql ready"
+  kubectl -n ${PRIMEHUB_NAMESPACE} rollout status deploy/primehub-graphql
+  info "[Progress] wait for console ready"
+  kubectl -n ${PRIMEHUB_NAMESPACE} rollout status deploy/primehub-console
+  info "[Progress] wait for helm install completed"
+}
+
+
+install::primehub() {
+  local PRIMEHUB_CONFIG=$(primehub_config_path)/helm_override/primehub.yaml
+  info "[Check] primehub.yaml"
+
+  if [ ! -f "${PRIMEHUB_CONFIG}" ]; then
+    mkdir -p $(primehub_config_path)/helm_override
+    info "[Generate] primehub.yaml"
+    if [[ $PRIMEHUB_MODE == 'ce' ]]; then
+      cp ${DIR}/helmfiles/primehub/helm_override/primehub-ce.yaml $PRIMEHUB_CONFIG
+    else
+      cp ${DIR}/helmfiles/primehub/helm_override/primehub-ee.yaml $PRIMEHUB_CONFIG
+    fi
+  else
+    info "[Found] ${PRIMEHUB_CONFIG}"
+  fi
+
+  info "[Install] PrimeHub"
+  export RELEASE_NAMESPACE=${PRIMEHUB_NAMESPACE}
+  export HELM_TIMEOUT=${HELM_TIMEOUT}
+  if [[ "${PRIMEHUB_CHART_PATH}" != "" ]]; then
+    export CHART_PATH=$(realpath "${PRIMEHUB_CHART_PATH}")
+  else
+    export CHART_VERSION=${PRIMEHUB_VERSION}
+  fi
+
+  trap 'kill $BGPID; exit' INT
+  progress_primehub &
+  BGPID=$!
+
```

Comment:
Add `helm repo update` before applying primehub.
Otherwise, helm can not fetch the latest PrimeHub version.

Reply:
Fixed. I use `sync` in replace of `charts`